.github/workflows/linting.yml

```
@@ -0,0 +1,15 @@
+name: pre-commit
+
+on:
+  pull_request:
+  push:
+    branches: [main]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: actions/setup-python@v2
+    - uses: pre-commit/action@v2.0.3
```

The following is from `ucal-workflows` PR for prefect3-pixi-upgrade - https://github.com/NSLS2/ucal-workflows/pull/1/:
```suggestion
    - uses: actions/checkout@v4.3.0
    - uses: actions/setup-python@v5.6.0
    - uses: pre-commit/action@v3.0.1
```